appinventor/appengine/src/com/google/appinventor/client/wizards/youngandroid/RemixedYoungAndroidProjectWizard.java

```diff
@@ -68,7 +68,7 @@ public void onKeyDown(KeyDownEvent event) {
     initFinishCommand(new Command() {
       @Override
       public void execute() {
-        String projectName = projectNameTextBox.getText();
+        String projectName = projectNameTextBox.getText().trim();
```

Comment:
@ewpatton 
Is this really required ? (adding trim to projectname)

Reply:
@Ramanathi My guess is that you will probably want it because otherwise your logic to replace spaces with underscores may result in project names either beginning or ending with an underscore if the user enters a name with a leading or trailing space.